.github/workflows/dotnet.yml

```diff
@@ -0,0 +1,51 @@
+name: build and test
+
+on:
+  push:   
+  pull_request:
+    branches: [ "main" ]   
+
+  
+jobs:
+  build:
+    strategy:
+      matrix:
+        configuration: [Debug]
+
+    runs-on: windows-latest                              
+
+    env:
+      Solution_Name: AnimalsTest.sln  
+      Project_Path: AnimalTest/AnimalTest/.csproj
+      Test_Project_Path: TestAnimals/TestAnimals/.csproj             
+     
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+  
+    - name: Install .NET 
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
+
+  
+    - name: Setup MSBuild.exe
+      uses: microsoft/setup-msbuild@v1.0.2      
+   
+    - name: Build
+      run: dotnet build
+   
+
+    - name: Test
+      run: dotnet test --no-restore --verbosity normal
+
+  
+    - name: Restore the application
```

Comment:
this step is not needed. Restoring packages should be done before the build or it will be done during the build. Repeating it one more time is not needed.

Reply:
If the intention was to run the app, `dotnet run` command should be used.